pyfarm/agent/utility.py

```diff
@@ -196,7 +196,11 @@ def dumps(*args, **kwargs):
     else:
         obj = dict(*args, **kwargs)
 
-    return _dumps(obj, default=default_json_encoder, indent=indent)
+    dump = _dumps(obj, default=default_json_encoder, indent=indent)
+
+    # Many parts of Twisted will not accept the typical output
+    # from _dumps, which is unicode, so we encode into ascii here.
+    return dump.encode("ascii", "replace")
```

Comment:
Agreed and I should have looked closer.  This turns out to be an artifact of the `DummyRequest` class which is forcing a check on the type of data being written.  I've fixed this in e13c5fc.